src/commands/yubihsm/keys/import.rs

```diff
@@ -13,115 +13,191 @@ pub struct ImportCommand {
     #[options(short = "c", long = "config")]
     pub config: Option<String>,
 
-    /// Path to the validator configuration file
-    #[options(short = "p", long = "path")]
-    pub path: Option<String>,
+    /// ID of the key to import (if applicable)
+    #[options(short = "i", long = "id")]
+    pub key_id: Option<u16>,
 
-    /// Type of key to import (default 'ed25519')
+    /// ID of the wrap key the original key was encrypted with
+    #[options(short = "w", long = "wrapkey")]
+    pub wrap_key_id: Option<u16>,
+
+    /// Type of key to import (either `wrap` or `priv_validator`, default `wrap`)
     #[options(short = "t")]
     pub key_type: Option<String>,
 
-    /// Label for imported key
+    /// Label for imported key (only applicable to `priv_validator` keys)
     #[options(short = "l", long = "label")]
     pub label: Option<String>,
 
-    /// Key ID for imported key
+    /// Path to the key to import
     #[options(free)]
-    key_id: Option<u16>,
+    pub path: PathBuf,
 }
 
 impl Callable for ImportCommand {
     fn call(&self) {
-        if self.path.is_none() {
-            status_err!("must provide a valid path to priv_validator.json");
+        let contents = fs::read_to_string(&self.path).unwrap_or_else(|e| {
+            status_err!("couldn't import file {}: {}", self.path.display(), e);
             process::exit(1);
+        });
+
```

Comment:
Some weird magic/guesswork/detection for what type of key to import if it isn't explicitly specified.

I think this should work right most of the time (and if it guesses wrong, it will only result in an error), however it'd be nice to eventually deprecate and remove the overloaded `priv_validator.json` import functionality, at least at such a time as people are no longer importing `priv_validator.json` keys in practice.